pkg/scheduler/scheduler.go

```diff
@@ -565,13 +565,13 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 			// Pod did not fit anywhere, so it is counted as a failure. If preemption
 			// succeeds, the pod should get counted as a success the next time we try to
 			// schedule it. (hopefully)
-			metrics.PodScheduleFailures.Inc()
+			prof.Metrics.PodUnschedulable(start)
```

Comment:
What about to just `metrics.ScheduleAttempts.WithLabelValues("unschedulable", pod.Spec.SchedulerName)`? Similarly for other two result cases. Resp.:
```
...
profile := pod.Spec.SchedulerName`
...
metrics.ScheduleAttempts.WithLabelValues("unschedulable", profile)
```

Reply:
Note that the function updates 2 metrics: the count and latency. Just repeating the code everywhere seems prone to errors.

Reply:
I would like to avoid any unnecessary wrappers around `metrics.XXX` unless it simplifies the code/makes it more transparent.

Reply:
I would say it makes sense for 2 reasons:

- There are 2 metrics that we don't want to forget to update
- They have more than one label, so a single location to control the order of the labels would help avoid ordering bugs.

Reply:
Oops, this last comment should have been for #92202

Reply:
https://github.com/kubernetes/kubernetes/pull/92202/files?file-filters%5B%5D=.go#diff-02b955ddc71dcbc4c68dc68bd62a5305 looks much better :)